### ISSUE TITLE
What is the use of the attribute 'state' in the RecEnv from the single item recommender tutorial?

### ISSUE THREAD

**Issue by davera-017** [facebookresearch/Pearl#97]
The environment class in this tutorial has an attribute called `state` which tracks the user history but apparently is never used. Here is its initialization code:

```python
class RecEnv(Environment):
    def __init__(
        self, actions: List[torch.Tensor], model: nn.Module, history_length: int
    ) -> None:
        ...
        self.state: torch.Tensor = torch.zeros((self.history_length, 100)).to(device)
        ...
```

I was wondering if this is somehow used inside other classes (e.g. some History Summarization Module) or if it is not used at all.

**Comment by rodrigodesalvobraz** [comment]
Hi Daniel,
Why do you say it is never used? When I search for `self.state`, I see:
![image](https://github.com/facebookresearch/Pearl/assets/12362395/e3c177d2-371f-47cc-8857-2e6e51d5d5b5)
The state is initialized with zeros. It is used to form `state_batch`, which is used to determine the reward in
```
reward = self.model(state_batch, action_batch) * 3
```
It is also gradually updated, upon each step, with the most recent action. After `history_length` steps, the zeros will have been overwritten and `self.state` it will be a record of the `history_length` last taken actions. This is the information used by `self.model` to compute the reward.
I hope that helps.
Rodrigo
